### PR TITLE
Ported proxy tensor tests over to core

### PR DESCRIPTION
**Pull request by Chillee**
Will fill out later

### CODE REVIEW

**Comment by Chillee** [comment]
@samdow Do we have any "mode-based dispatch" tests?

**Comment by ezyang** [comment]
did you black proxy_tensor lol. Can you do it in a separate commit

**Comment by Chillee** [comment]
@ezyang Yeah. Just changing 2 things in `proxy_tensor`.

File: test/test_proxy_tensor.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# Owner(s): ["oncall: fx"]
```
Comment (samdow) [inline test/test_proxy_tensor.py:1]: Might be good to have these and the tests from `test_fx_experimental` in one file instead of two

**Review by ezyang — APPROVED** [review]

File: test/test_proxy_tensor.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+try:
+    import torchvision
+    USE_TORCHVISION = True
+except ImportError:
+    warnings.warn("Couldn't import torchvision. Some of our tests use it, try "
+                  "to install it with commands from pytorch.org, post-fixed with "
+                  "`--no-deps` to avoid overwriting the pytorch installation",
+                  UserWarning)
+
+# NB: numpy is a testing dependency!
```
Comment (zou3519) [inline test/test_proxy_tensor.py:60]: What does this mean? There's no usage of numpy in this function

File: test/test_proxy_tensor.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            kwargs = sample_input.kwargs
+
+            new_f = make_fx(f)(args, kwargs)
+            for arg in args:
+                if isinstance(arg, torch.Tensor) and arg.dtype == torch.float:
+                    arg.uniform_(0, 1)
+            try:
+                old_out = f(args, kwargs)
+            except Exception:
+                continue
```
Comment (zou3519) [inline test/test_proxy_tensor.py:196]: seems unsafe

File: test/test_proxy_tensor.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+            new_out = new_f(args, kwargs)
+            self.assertEqual(new_out, old_out)
+            pass
```
Comment (zou3519) [inline test/test_proxy_tensor.py:199]: remove pass?

File: test/test_proxy_tensor.py [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+USE_TORCHVISION = False
+try:
+    import torchvision
+    USE_TORCHVISION = True
+except ImportError:
+    warnings.warn("Couldn't import torchvision. Some of our tests use it, try "
+                  "to install it with commands from pytorch.org, post-fixed with "
+                  "`--no-deps` to avoid overwriting the pytorch installation",
+                  UserWarning)
```
Comment (zou3519) [inline test/test_proxy_tensor.py:58]: Does PyTorch CI install torchvision? this might never get hit

**Review by zou3519 — APPROVED** [review]

File: test/test_proxy_tensor.py (hunk 3 above)
Comment (Chillee) [reply test/test_proxy_tensor.py:196]: It's meant as an universal check against inputs that aren't valid.

File: test/test_proxy_tensor.py (hunk 5 above)
Comment (Chillee) [reply test/test_proxy_tensor.py:58]: I think other tests in PyTorch core do have a line like this, so I think it's reasonable.

**Pull request closed without merging**